hydra-cluster/test/Test/HydraExplorerSpec.hs

```diff
@@ -0,0 +1,239 @@
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \cardanoNode@RunningNode{nodeSocket} -> do
+            let hydraTracer = contramap FromHydraNode tracer
+            hydraScriptsTxId <- publishHydraScriptsAs cardanoNode Faucet
+            (aliceCardanoVk, _aliceCardanoSk) <- keysFor Alice
+            aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket hydraScriptsTxId [] cperiod
+            withHydraNode hydraTracer aliceChainConfig tmpDir 1 aliceSk [] [1] $ \hydraNode -> do
+              withHydraExplorer cardanoNode $ \explorer -> do
+                seedFromFaucet_ cardanoNode aliceCardanoVk 100_000_000 (contramap FromFaucet tracer)
+
+                send hydraNode $ input "Init" []
+
+                headId <- waitMatch 5 hydraNode $ \v -> do
+                  guard $ v ^? key "tag" == Just "HeadIsInitializing"
+                  v ^? key "headId" . _String
+
+                headExplorerSees explorer "HeadInitTx" headId
+
+                requestCommitTx hydraNode mempty >>= submitTx cardanoNode
+                waitFor hydraTracer 5 [hydraNode] $
+                  output "HeadIsOpen" ["utxo" .= object mempty, "headId" .= headId]
+
+                headExplorerSees explorer "HeadCommitTx" headId
+                headExplorerSees explorer "HeadCollectComTx" headId
+
+                send hydraNode $ input "Close" []
+
+                headExplorerSees explorer "HeadCloseTx" headId
+
+                waitFor hydraTracer 50 [hydraNode] $
+                  output "ReadyToFanout" ["headId" .= headId]
+
+                send hydraNode $ input "Fanout" []
+
+                headExplorerSees explorer "HeadFanoutTx" headId
+
+  it "can observe hydra transactions created by multiple hydra-nodes" $
+    failAfter 60 $
+      showLogsOnFailure "HydraExplorerSpec" $ \tracer -> do
+        withTempDir "hydra-explorer-history" $ \tmpDir -> do
+          withCardanoNodeDevnet (contramap FromCardanoNode tracer) tmpDir $ \cardanoNode@RunningNode{nodeSocket} -> do
+            let hydraTracer = contramap FromHydraNode tracer
+            hydraScriptsTxId <- publishHydraScriptsAs cardanoNode Faucet
+
+            let initHead hydraNode = do
+                  send hydraNode $ input "Init" []
+                  waitMatch 5 hydraNode $ \v -> do
+                    guard $ v ^? key "tag" == Just "HeadIsInitializing"
+                    v ^? key "headId" . _String
+
+            (aliceCardanoVk, _aliceCardanoSk) <- keysFor Alice
+            seedFromFaucet_ cardanoNode aliceCardanoVk 100_000_000 (contramap FromFaucet tracer)
+            aliceChainConfig <- chainConfigFor Alice tmpDir nodeSocket hydraScriptsTxId [] cperiod
+            aliceHeadId <- withHydraNode hydraTracer aliceChainConfig tmpDir 1 aliceSk [] [1] initHead
+
+            (bobCardanoVk, _bobCardanoSk) <- keysFor Bob
+            seedFromFaucet_ cardanoNode bobCardanoVk 100_000_000 (contramap FromFaucet tracer)
```

Comment:
The seed ada amount of 100 is probably a bit too much especially since we only init and finish the test.